src/realm.js

```diff
@@ -656,21 +656,22 @@ export class Realm {
     return new Constructor(this, types, values, args, buildNode, { kind, intrinsicName });
   }
 
-  rebuildObjectProperty(object: Value, key: string, propertyValue: Value, path: string) {
+  rebuildObjectProperty(generator: Generator, object: Value, key: string, propertyValue: Value, path: string) {
     if (!(propertyValue instanceof AbstractValue)) return;
     if (!propertyValue.isIntrinsic()) {
       propertyValue.intrinsicName = `${path}.${key}`;
       propertyValue.args = [object];
       propertyValue._buildNode = ([node]) => t.memberExpression(node, t.identifier(key));
-      this.rebuildNestedProperties(propertyValue, propertyValue.intrinsicName);
+      this.rebuildNestedProperties(generator, propertyValue, propertyValue.intrinsicName);
     }
   }
 
-  rebuildNestedProperties(abstractValue: AbstractValue | UndefinedValue, path: string) {
+  rebuildNestedProperties(generator: Generator, abstractValue: AbstractValue | UndefinedValue, path: string) {
     if (!(abstractValue instanceof AbstractObjectValue)) return;
     let template = abstractValue.getTemplate();
     invariant(!template.intrinsicName || template.intrinsicName === path);
     template.intrinsicName = path;
+    template.generator = generator;
```

Comment:
It took me a while before I could satisfy myself that template.generator is only assigned to once. Perhaps a comment and an invariant would help with this.